scripts/ioc_deploy.py

```diff
@@ -180,35 +220,61 @@ def get_parser(subparser: bool = False) -> argparse.ArgumentParser:
             "--yes",
             "-y",
             action="store_true",
+            default=argparse.SUPPRESS,
             help="Skip the confirmation promps, automatically saying yes to each one.",
         )
         parser.add_argument(
             "--dry-run",
             action="store_true",
+            default=argparse.SUPPRESS,
             help="Do not deploy anything, just print what would have been done.",
         )
         parser.add_argument(
             "--verbose",
             "-v",
             "--debug",
             action="store_true",
+            default=argparse.SUPPRESS,
             help="Display additional debug information.",
         )
     # main_parser unique arguments that should go last
     main_parser.add_argument(
         "--github_org",
         "--org",
         action="store",
-        default=current_default_org,
+        default=DEFAULT_ARGS.github_org,
         help=(
             "The github org to deploy IOCs from. "
             f"This defaults to $GITHUB_ORG, or {GITHUB_ORG_DEFAULT} if the environment variable is not set. "
-            f"With your current environment variables, this defaults to {current_default_org}."
+            f"With your current environment variables, this defaults to {DEFAULT_ARGS.github_org}."
         ),
     )
-    if subparser:
+    if not subparser:
+        return main_parser
+    elif subparser == PERMS_CMD:
         return perms_parser
-    return main_parser
+    elif subparser == REBUILD_CMD:
+        return rebuild_parser
+    raise ValueError(f"Subparser argument must be empty string or one of {ALL_SUBCOMMANDS}, not {subparser}")
+
+
+def args_to_dataclass(args: argparse.Namespace) -> CliArgs:
+    """
+    Interpret the results of argparse as a CliArgs dataclass.
+
+    Unlike a standard argparse parsing, we could be missing
+    arguments entirely from the namespace via argparse.SUPPRESS.
+
+    This is required because our subparsers share arguments
+    with the main parser, and if they have default values
+    they will override the main parser every time.
+
+    Therefore, it's the duty of this function to fill in any
+    missing arguments with their default values.
+    """
+    kw = vars(DEFAULT_ARGS)
+    kw.update(vars(args))
```

Comment:
Nit: I think [`dataclass.replace()`](https://docs.python.org/3/library/dataclasses.html#dataclasses.replace) could work here.  Not that this is that much more verbose.

Reply:
til dataclass.replace

Reply:
forgot that I'm still supporting rhel7 installed python3 which doesn't have dataclasses yet
putting this in as a comment instead of as a change